miasm2/analysis/mem.py

```diff
@@ -0,0 +1,1259 @@
+
+
+# FIXME: copied from miasm2.os_dep.common and fixed
+def get_str_ansi(vm, addr, max_char=None):
+    """Get a null terminated ANSI encoded string from a VmMngr.
+
+    Args:
+        vm: VmMngr instance
+        max_char: max number of characters to get in memory
+    """
+    l = 0
+    tmp = addr
+    while ((max_char is None or l < max_char) and
+           vm.get_mem(tmp, 1) != "\x00"):
+        tmp += 1
+        l += 1
+    return vm.get_mem(addr, l).decode("latin1")
+
+
+# TODO: get_raw_str_utf16 for length calculus
+def get_str_utf16(vm, addr, max_char=None):
+    """Get a (double) null terminated utf16 little endian encoded string from
+    a VmMngr. This encoding is mainly used in Windows.
+
+    FIXME: the implementation do not work with codepoints that are encoded on
+    more than 2 bytes in utf16.
+
+    Args:
+        vm: VmMngr instance
+        max_char: max number of bytes to get in memory
+    """
+    l = 0
+    tmp = addr
+    # TODO: test if fetching per page rather than 2 byte per 2 byte is worth it?
+    while ((max_char is None or l < max_char) and
+           vm.get_mem(tmp, 2) != "\x00\x00"):
+        tmp += 2
+        l += 2
+    s = vm.get_mem(addr, l)
+    return s.decode('utf-16le')
+
+
+def set_str_ansi(vm, addr, s):
+    """Encode a string to null terminated ascii/ansi and set it in a VmMngr
+    memory.
+
+    Args:
+        vm: VmMngr instance
+        addr: start address to serialize the string to
+        s: the str to serialize
+    """
+    vm.set_mem(addr, s + "\x00")
+
+
+def set_str_utf16(vm, addr, s):
+    """Same as set_str_ansi with (double) null terminated utf16 encoding."""
+    s = (s + '\x00').encode('utf-16le')
+    vm.set_mem(addr, s)
+
+
+# MemField to MemStruct helper
+
+def mem(field):
+    """Generate a MemStruct subclass from a field. The field's value can
+    be accessed through self.value or self.deref_value if field is a Ptr.
+    """
+    if field in DYN_MEM_STRUCT_CACHE:
+        return DYN_MEM_STRUCT_CACHE[field]
+
+    fields = [("value", field)]
+    # Build a type to contain the field type
+    mem_type = type("Mem%r" % field, (MemStruct,), {'fields': fields})
+    DYN_MEM_STRUCT_CACHE[field] = mem_type
+    return mem_type
+
+
+# MemField classes
+
+class MemField(object):
+    """Base class to provide methods to set and get fields from virtual mem.
+
+    Subclasses can either override _pack and _unpack, or get and set if data
+    serialization requires more work (see Inline implementation for an example).
+    """
+
+    _self_type = None
+
+    def _pack(self, val):
+        """Serializes the python value @val to a raw str"""
+        raise NotImplementedError()
+
+    def _unpack(self, raw_str):
+        """Deserializes a raw str to an object representing the python value
+        of this field.
+        """
+        raise NotImplementedError()
+
+    def set(self, vm, addr, val):
+        """Set a VmMngr memory from a value.
+
+        Args:
```

Comment:
To be more homogeneous with other docstring in Miasm, you can use the following format :
```Python
"""Set a VmMngr memory from a value
@vm: VmMngr instance
@addr: the start adress in memory to set
@val: the python value to serialize in @vm at @addr
"""
```